h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstFillNA.java

```diff
@@ -98,6 +105,41 @@ public void map(Chunk cs[], NewChunk nc[]) {
       }
     }
   }
+
+  /***
+   * This class was shameless copied from FillForwardTaskRow from Pavel.  We gain speed here by not having
```

Comment:
I'd remove the first sentence. No need for apologies and the FillForwardTaskRow is not written by me at all. 

The **code duplication is ok**, as speed gain is what we want when iterating through chunks.Not really a duplication. In theory, algorithms should be general. In practice, we need them to finish ASAP :)